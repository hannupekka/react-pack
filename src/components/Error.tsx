import styles from '@app/styles/components/Error.less';
import * as React from 'react';
import { pure } from 'recompose';

interface Props {
  message: string;
}

const Error = (props: Props) => <div className={styles.error}>{props.message}</div>;

export default pure(Error);
