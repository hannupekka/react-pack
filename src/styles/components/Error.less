@import '~styles/main';

.error {
  border: 1px solid @color__red;
  background-color: @color__pink;
  padding: @padding;
}
