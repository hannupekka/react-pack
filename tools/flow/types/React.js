declare type ElementType = React$Element<any>
