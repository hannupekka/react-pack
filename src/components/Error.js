// @flow
import styles from 'styles/components/Error';
import React from 'react';
import { pure } from 'recompose';
import type { Component } from 'recompose';
import CSSModules from 'react-css-modules';

type Props = {
  message: string
}

const Error: Component<Props> = (props: Props): ElementType => {
  return (
    <div styleName="error">
      {props.message}
    </div>
  );
};

export default pure(CSSModules(Error, styles));
