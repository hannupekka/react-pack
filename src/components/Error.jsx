import styles from 'styles/components/Error.less';
import React from 'react';
import PropTypes from 'prop-types';
import CSSModules from 'react-css-modules';

const Error = props => <div styleName="error">{props.message}</div>;

Error.propTypes = {
  message: PropTypes.string,
};

Error.defaultProps = {
  message: '',
};

export default CSSModules(Error, styles);
