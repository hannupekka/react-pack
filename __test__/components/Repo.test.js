import React from 'react';
import renderer from 'react-test-renderer';
import Error from 'components/Error';

describe('Error', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Error message="Error!" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
