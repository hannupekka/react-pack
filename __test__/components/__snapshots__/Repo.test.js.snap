// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders correctly 1`] = `
<div
  className="error"
>
  Error!
</div>
`;
